### DISCUSSION TITLE
A working example of a server that uses shelf

### DISCUSSION THREAD

**Discussion by ramezanifar** [FreeOpcUa/python-opcua#1526] [Q&A]
Can someone share a working sample that uses a shelf file during both creation and more importantly loading?
In an old issue conversation, I see there is a mention of "sheffile" argument in the server() method. But in the asyncua module that I use, that input argument does not exist. I see that the init() method has and input argument named "shelf_file".
When I assign a file name to it and run, it takes longer than usual and I see some files are created with the extensions "bak", "dat", and "dir".
But what is next? How should I reuse that? 
I put the  logic for node creation in an if condition to bypass them for the second run. But after initialization, in my opc  client viewer (such as uaexpert) I don't see any of the nodes. 
A workable example would be helpful.
Thank you

**Comment by schroeder-** [comment]
Shelf file isn't support anymore. But you can export your address space as xml and then load it in the next run.
